#! /bin/sh -e
## DP: 03_set-umask.diff
if [ $# -ne 1 ]; then 
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1  
fi
case "$1" in 
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

@DPATCH@
--- mylvmbackup-0.15.orig/mylvmbackup.pl.in	2011-02-14 14:04:25.000000000 +0100
+++ mylvmbackup-0.15/mylvmbackup.pl.in	2011-02-14 14:04:31.000000000 +0100
@@ -510,6 +510,7 @@
 {
   my $tarball = $archivename.$tarfilesuffix;
   my $tarballtmp = mktemp("$tarball.INCOMPLETE-XXXXXXX");
+  umask 077;
 
   log_msg ("Taking actual backup...", LOG_INFO);
   log_msg ("Creating tar archive $tarball", LOG_INFO);
